subprojects/native/src/main/java/org/gradle/internal/nativeintegration/filesystem/jdk7/NioFileMetadataAccessor.java

```diff
@@ -16,29 +16,41 @@
 package org.gradle.internal.nativeintegration.filesystem.jdk7;
 
 import org.gradle.internal.file.FileMetadataSnapshot;
+import org.gradle.internal.file.FileMetadataSnapshot.AccessType;
 import org.gradle.internal.file.impl.DefaultFileMetadataSnapshot;
 import org.gradle.internal.nativeintegration.filesystem.FileMetadataAccessor;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
 import java.nio.file.attribute.BasicFileAttributes;
 
 @SuppressWarnings("Since15")
 public class NioFileMetadataAccessor implements FileMetadataAccessor {
     @Override
     public FileMetadataSnapshot stat(File f) {
         try {
-            BasicFileAttributes bfa = Files.readAttributes(f.toPath(), BasicFileAttributes.class);
+            Path path = f.toPath();
+            BasicFileAttributes bfa = java.nio.file.Files.readAttributes(path, BasicFileAttributes.class, LinkOption.NOFOLLOW_LINKS);
+            AccessType accessType = bfa.isSymbolicLink() ? AccessType.VIA_SYMLINK : AccessType.DIRECT;
+            if (accessType == AccessType.VIA_SYMLINK) {
+                try {
+                    bfa = Files.readAttributes(path, BasicFileAttributes.class);
+                } catch (IOException e) {
```

Comment:
We should catch `java.nio.file.NoSuchFileException` here.

Reply:
We should not, see above.